Fall back to full view if activeTabID or relevant pages are not found

canova opened the pull request:
Fixes #3386. 
You can reproduce it manually with running `MOZ_PROFILER_STARTUP=1` and capturing the profile on release.
It's hard to reproduce it otherwise but I've addes some tests that were failing before.

julienw commented on src/test/store/active-tab.test.js:
```diff
@@ -111,6 +120,15 @@ describe('finalizeProfileView', function() {
+      profile.meta.configuration = {
+        ...profile.meta.configuration,
+        // null is represented by 0 for activeTab in the back-end.
+        activeTabID: activeTabID ?? 0,
```
😁

julienw commented on src/actions/receive-profile.js:
```diff
@@ -184,6 +185,7 @@ export function finalizeProfileView(
     // encoded into the URL.
     const selectedThreadIndexes = getSelectedThreadIndexesOrNull(getState());
     const pages = profile.pages;
+    const activeTabID = getActiveTabID(getState());
```
super nit: this is used only in the block for `case 'active-tab':`, can you please define it there? That will make it easier to read when we get it closer to its use.
Note you'll need to put brackets around the case block so that eslint doesn't yell at you.

julienw commented on src/actions/receive-profile.js:
```diff
@@ -197,7 +199,7 @@ export function finalizeProfileView(
         break;
       case 'active-tab':
-        if (pages) {
+        if (pages && activeTabID !== null) {
```
It would be good to fully document the cases where this can happen.

julienw reviewed (approved):
Thanks, this seems to fix the problem on my side too!

Do we need to add `| null` here ? https://github.com/firefox-devtools/profiler/blob/c4a70c212bcb17a91a5a6f794c8983291f87b9e1/src/types/profile.js#L503

So, what is the underlying issue? Why don't we have this active tab id in this case?
My understanding is that because of startup profiling there's no "active tab id" information, and in release the "web developer" preset is the default. (I could reproduce with a Nightly version, after I manually set the "Web Developer" preset).

Should we document this more?

Thanks :-)

canova commented:
Thanks for the quick review!

> Do we need to add `| null` here ?
>
> https://github.com/firefox-devtools/profiler/blob/c4a70c212bcb17a91a5a6f794c8983291f87b9e1/src/types/profile.js#L503

Nope, Firefox doesn't output `null`, it outputs `0` as null value. We are computing this while getting with `getActiveTabID` selector:
https://github.com/firefox-devtools/profiler/blob/9b4c7ad625e03dacb253b9e4749226197dfa7dcc/src/selectors/profile.js#L229-L242
I added a comment explaining this there.

> So, what is the underlying issue? Why don't we have this active tab id in this case?
> My understanding is that because of startup profiling there's no "active tab id" information, and in release the "web developer" preset is the default. (I could reproduce with a Nightly version, after I manually set the "Web Developer" preset).

Yeah, when the preset is "Web Developer" is selected, it tries to open that as a web dev mode. But since we do a startup profiling, it's not possible to get the activeTabID, so it passes `0` as null there. 
 
> Should we document this more?

Yeah, not sure where though. I added a comment to explain this zero value for the activeTabID. Any more ideas?
